### PR TITLE
State knowledge fixes

### PR DESCRIPTION
**Pull request by leohtkam**
Changed state knowledge implementation to use global variable to track what state the user is in. Also changed survey summary question edit to reuse the survey question page.

### CODE REVIEW

**Comment by impguard** [comment]
+1 if it works. I didn't actual test it just looked at code.

**Comment by whsieh** [comment]
^ can confirm, everything works.

File: app/assets/javascripts/startpage/surveyTopics.coffee [hunk 1]
```diff
@@ -10,11 +10,15 @@ surveyTopics.config(["$routeProvider", ($routeProvider) ->
     $scope.MIN_NUM_TOPICS_REQUIRED = 5
     $scope.allTopics = []
     $scope.topicSelectionModel = {}
-    
+
+    currentState = StartPageData.getCurrentState()
```
Comment (whsieh) [inline app/assets/javascripts/startpage/surveyTopics.coffee:14]: more concise: currentState = StartPageData.getCurrentState() || "topics"

**Comment by whsieh** [comment]
+1, let's merge it in

**Pull request merged**